llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -3876,10 +3876,15 @@ static SDValue getAArch64Cmp(SDValue LHS, SDValue RHS, ISD::CondCode CC,
   //    cmp     w13, w12
   // can be turned into:
   //    cmp     w12, w11, lsl #1
-  if (!isa<ConstantSDNode>(RHS) || !isLegalArithImmed(RHS->getAsZExtVal())) {
-    SDValue TheLHS = isCMN(LHS, CC) ? LHS.getOperand(1) : LHS;
-
-    if (getCmpOperandFoldingProfit(TheLHS) > getCmpOperandFoldingProfit(RHS)) {
+  if (!isa<ConstantSDNode>(RHS) ||
+      !isLegalArithImmed(RHS->getAsAPIntVal().abs().getZExtValue())) {
+    bool LHSIsCMN = LHS.getOpcode() == ISD::SUB && isCMN(LHS, CC);
+    bool RHSIsCMN = RHS.getOpcode() == ISD::SUB && isCMN(RHS, CC);
```

Comment:
Answer: I should remove that from isCMN because when I removed the other SUB, LLVM would crash as not all nodes have multiple operands

Reply:
Sorry, I'm not sure what mean. I don't see isCMN removing the `Op.getOpcode() == ISD::SUB` in #96349, and it would seem like a redundant check. Can you explain in more detail what would crash?

Reply:
The passing of the getOperand(1) as an argument. 

getOperand will assert if there's not enough operands, causing a crash.

Reply:
I see. Can we add that later, once it is needed?

Reply:
> I see. Can we add that later, once it is needed?

Alright!